pkg/controllers/resources/persistentvolumes/syncer.go

```diff
@@ -192,16 +163,43 @@ func (s *persistentVolumeSyncer) Sync(ctx *synccontext.SyncContext, pObj client.
 			}
 			return ctrl.Result{}, err
 		}
+	}
 
-		updatedPv := s.translateUpdate(ctx, vPersistentVolume, pPersistentVolume)
-		if updatedPv != nil {
-			ctx.Log.Infof("update physical persistent volume %s, because spec or annotations have changed", updatedPv.Name)
-			translator.PrintChanges(pPersistentVolume, updatedPv, ctx.Log)
-			err := ctx.PhysicalClient.Update(ctx, updatedPv)
-			if err != nil {
-				return ctrl.Result{}, err
-			}
+	// patch objects
+	patch, err := patcher.NewSyncerPatcher(ctx, pPersistentVolume, vPersistentVolume)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("new syncer patcher: %w", err)
+	}
+	defer func() {
+		if err := patch.Patch(ctx, pPersistentVolume, vPersistentVolume); err != nil {
+			retErr = utilerrors.NewAggregate([]error{retErr, err})
 		}
+	}()
+
+	// bidirectional update
+	sourceObject, targetObject := synccontext.SyncSourceTarget(ctx, pPersistentVolume, vPersistentVolume)
+	targetObject.Spec.PersistentVolumeSource = sourceObject.Spec.PersistentVolumeSource
+	targetObject.Spec.Capacity = sourceObject.Spec.Capacity
+	targetObject.Spec.AccessModes = sourceObject.Spec.AccessModes
+	targetObject.Spec.PersistentVolumeReclaimPolicy = sourceObject.Spec.PersistentVolumeReclaimPolicy
+	targetObject.Spec.NodeAffinity = sourceObject.Spec.NodeAffinity
+	targetObject.Spec.VolumeMode = sourceObject.Spec.VolumeMode
+	targetObject.Spec.MountOptions = sourceObject.Spec.MountOptions
+
+	// update virtual object
+	err = s.translateUpdateBackwards(ctx, vPersistentVolume, pPersistentVolume, vPvc)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// update virtual status
+	vPersistentVolume.Status = *pPersistentVolume.Status.DeepCopy()
```

Comment:
why do we do a deepcopy here?

Reply:
Yeah I guess its not necessary I can remove that if you want